python/torch_mlir/eager_mode/torch_mlir_dispatch.py

```diff
@@ -0,0 +1,258 @@
+# Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+# Also available under a BSD-style license. See LICENSE.
+
+import re
+import warnings
+from typing import Any
+from typing import List, Dict
+
+import numpy as np
+import torch
+import torch._C
+from torch.utils._pytree import tree_map
+from torch_mlir._mlir_libs._mlir.passmanager import PassManager
+
+from torch_mlir.dialects.torch.importer.jit_ir.build_tools.registry import Registry
+from torch_mlir.eager_mode.ir_building import build_module
+from torch_mlir.dialects import torch as torch_dialect
+from torch_mlir.eager_mode.torch_mlir_types import TorchTensorType
+
+OP_REGISTRY = Registry.load()
+SUPPORTED_OPS = frozenset(
+    [
+        member.OPERATION_NAME
+        for member in vars(torch_dialect).values()
+        if hasattr(member, "OPERATION_NAME")
+    ]
+)
+
+
+def check_supported_op(schema: torch._C.FunctionSchema) -> bool:
+    return (
+        "torch."
+        + schema.name.replace("::", ".")
+        + ("." + schema.overload_name if schema.overload_name else "")
+    ) in SUPPORTED_OPS
+
+
+# shameless SO copy pasta
```

Comment:
Please, no copypasta! That is bad for licensing reasons.

Note that MLIR allows arbitrary strings as function names, so you shouldn't need to munge this at all. (can just remove this code)